synapse/replication/tcp/client.py

```diff
@@ -141,21 +141,22 @@ async def on_rdata(
                 if row.type != EventsStreamEventRow.TypeId:
                     continue
                 assert isinstance(row, EventsStreamRow)
-
-                event = await self.store.get_event(
-                    row.data.event_id, allow_rejected=True
-                )
-                if event.rejected_reason:
-                    continue
```

Comment:
Should we be checking `row.data.rejected`? Not sure if dropping this clause was done on purpose.

Reply:
Well that's embarrassing.